File: leetcode/Math/117-Excel_Sheet_Column_Number/sol.py
class Solution:
    # @param {string} s
    # @return {integer}
    def titleToNumber(self, s):
    	l = len(s);
        return  (ord(s[0])-64) if (l==1) else (ord(s[l-1])-64) + 26*self.titleToNumber(s[:-1]);

# Test Case
if __name__=="__main__":
    ori = "ABC";
    cla = Solution()
    print cla.titleToNumber(ori)

File: leetcode/Math/117-Excel_Sheet_Column_Number/sol.c
/**
 * @Time Complexity:	O(n)
 * @Space Complexity:	O(2n)
 */
#include <string.h>
#include <stdio.h>

/* Solution */
int titleToNumber(char* s) {
	int ret = 0;
	for (int i = 0; i < strlen(s); i++) {
		ret = ret * 26 + s[i] - 64;
	}
	return ret;
}

/* Test Case */
int main() {
	char in[10] = "AA";
	printf("%d\n", titleToNumber(in));
}
